utils/bin/poller/src/main.rs

```diff
@@ -0,0 +1,301 @@
+    // This is needed so that we can lookup an oracle, unless we otherwise have the price to
+    // supply.
+    let data = helper
+        .getSnapshot(amm)
+        .state(overrides.clone())
+        .call()
+        .await?;
+
+    // Check if the AMM is legacy
+    let legacy = !data._0.is_empty();
+
+    let numerator = std::env::var("NUMERATOR");
+    let denominator = std::env::var("DENOMINATOR");
+
+    let (numerator, denominator, other_post_interactions) = match legacy {
+        true => {
+            // Third decode the data and get the trading params
+            let (data,) =
+                ComposableCoW::ConditionalOrderCreated::abi_decode_data(&data._0, true).unwrap();
+            let trading_params = LegacyTradingParams::abi_decode(&data.staticInput, true).unwrap();
+
+            let (numerator, denominator) = match (numerator, denominator) {
+                (Ok(n), Ok(d)) => (U256::from_str(&n)?, U256::from_str(&d)?),
+                _ => {
+                    let oracle_price =
+                        IPriceOracle::new(trading_params.priceOracle, provider.clone())
+                            .getPrice(
+                                trading_params.token0,
+                                trading_params.token1,
+                                trading_params.priceOracleData,
+                            )
+                            .call()
+                            .await
+                            .unwrap();
+                    println!(
+                        "Oracle Price: {:?} / {:?}",
+                        oracle_price.priceNumerator, oracle_price.priceDenominator
+                    );
+                    (oracle_price.priceNumerator, oracle_price.priceDenominator)
+                }
+            };
+
+            (
+                numerator,
+                denominator,
+                Some(IMulticall3::Call {
+                    target: data.handler,
+                    callData: LegacyConstantProduct::commitmentCall { amm }
+                        .abi_encode()
+                        .into(),
+                }),
+            )
+        }
+        false => match (numerator, denominator) {
+            (Ok(n), Ok(d)) => (U256::from_str(&n)?, U256::from_str(&d)?, None),
+            _ => {
+                return Err(eyre::eyre!(
+                    "NUMERATOR and DENOMINATOR env vars must be set for non-legacy AMMs"
+                ))
+            }
+        },
+    };
+
+    // Fifth, we now have relative prices, so we can use the helper to get the order
+    let hint = helper
+        .order(amm, vec![numerator, denominator])
+        .state(overrides)
+        .call_raw()
+        .await
+        .map_or_else(Err, |d| {
+            let dReturn {
+                order,
+                preInteractions,
+                postInteractions,
+                sig,
+            } = IConstantProductHelper::dCall::abi_decode_returns(&d, true).unwrap();
```

Comment:
Library question, unrelated to this PR. What does `abi_decode_return(_, true)` mean, specifically the `true`? From the [docs](https://alloy-rs.github.io/alloy/alloy/dyn_abi/trait.FunctionExt.html#tymethod.abi_decode_output) I see that the parameter name is `validate`, but it's unclear what it's validating or why this isn't just `abi_decode_return(_)`.